lib/compilers/java.js

```diff
@@ -124,8 +127,36 @@ export class JavaCompiler extends BaseCompiler {
         return result;
     }
 
-    getMainClassName() {
-        // TODO(supergrecko): The main class name is currently hardcoded
+    async getMainClassName(dirPath) {
+        const maxSize = this.env.ceProps('max-asm-size', 64 * 1024 * 1024);
+        const files = await fs.readdir(dirPath);
+        const results = await Promise.all(
+            files
+                .filter(f => f.endsWith('.class'))
+                .map(async classFile => {
+                    const options = {
+                        maxOutput: maxSize,
+                        customCwd: dirPath,
+                    };
+                    const objResult = await this.exec(this.compiler.objdumper, [classFile], options);
+                    if (objResult.code !== 0) {
+                        return null;
```

Comment:
Are there any options/differences that prevent us from reusing the objdump method?

Reply:
As I understand, the API can receive execution-only requests which wouldn't need a call to objdump, furthermore by separating these calls out we can chose to call `javap` without any of the byte code options. This would make the checks for `main(` have to include detection for possible string literals which could contain `public static void main(...`

Reply:
That's fair